diff --git a/src/plugins/_api/badges/index.tsx b/src/plugins/_api/badges/index.tsx
index aa8111a..7014cad 100644
--- a/src/plugins/_api/badges/index.tsx
+++ b/src/plugins/_api/badges/index.tsx
@@ -94,11 +94,8 @@ async function loadBadges(url: string, noCache = false) {
 }
 
 async function loadAllBadges(noCache = false) {
-    const vencordBadges = await loadBadges("https://badges.vencord.dev/badges.json", noCache);
-    const equicordBadges = await loadBadges("https://badge.equicord.org/badges.json", noCache);
-
-    DonorBadges = vencordBadges;
-    EquicordDonorBadges = equicordBadges;
+    DonorBadges = {};
+    EquicordDonorBadges = {};
 }
 
 let intervalId: any;
