diff --git a/src/webpack/common/utils.ts b/src/webpack/common/utils.ts
index 4790315..8f82f28 100644
--- a/src/webpack/common/utils.ts
+++ b/src/webpack/common/utils.ts
@@ -130,9 +130,7 @@ export const ChannelRouter: t.ChannelRouter = mapMangledModuleLazy('"Thread must
     transitionToThread: filters.byCode('"Thread must have a parent ID."')
 });
 
-export const SettingsRouter: t.SettingsRouter = mapMangledModuleLazy('type:"USER_SETTINGS_MODAL_OPEN"', {
-    openUserSettings: filters.byCode('type:"USER_SETTINGS_MODAL_OPEN"')
-});
+export const SettingsRouter: t.SettingsRouter = findByPropsLazy("openUserSettings");
 
 export const PermissionsBits: t.PermissionsBits = findLazy(m => typeof m.ADMINISTRATOR === "bigint");
 
